### ISSUE TITLE
Inconsistent file tree depending on OS

### ISSUE THREAD

**Issue by adrien-febvay** [sindresorhus/cpy-cli#47]
Consider the following file tree:
```
dist-tmp
└── index.html
```

On Windows, the command `npx cpy-cli dist-tmp/** dist/` results in:
```
dist
└── index.html
```

On Linux, the same command `npx cpy-cli dist-tmp/** dist/` results in:
```
dist
└── dist-tmp
    └── index.html
```

It happened using node 18.18.0 on both systems

**Comment by yukha-dw** [comment]
Using this command should give same results `cpy dist-tmp/**/* dist/`
```
dist
└── index.html
```

**Comment by adrien-febvay** [comment]
> Using this command should give same results `cpy dist-tmp/**/* dist/`

No, it results in `'cpy' is not recognized as an internal or external command, operable program or batch file.`

One of the point of this package is having a cross-platform solution.

**Comment by zhangyahan** [comment]
I also have the same problem

**Comment by yukha-dw** [comment]
> > Using this command should give same results `cpy dist-tmp/**/* dist/`
> 
> No, it results in `'cpy' is not recognized as an internal or external command, operable program or batch file.`
> 
> One of the point of this package is having a cross-platform solution.

replace `cpy` with `npx cpy-cli` if executed directly

**Comment by adrien-febvay** [comment]
> replace `cpy` with `npx cpy-cli` if executed directly

The point of this thread is that `npx cpy-cli` has an inconsistent behaviour depending on the OS you're on, so your answer telling to use it doesn't make any sense as, once again, it is broken.

Anyway, I just found out that adding `/*` to the source fixes my issue.

My entire command becomes `npx cpy-cli dist-tmp/**/* dist/`
instead of just `npx cpy-cli dist-tmp/** dist/`

While my issue is resolved, having an inconsistent behaviour is still problematic. It should be fixed, and until then my workaround should be documented.

**Comment by sanjaikumarv** [comment]
Same issue
    "copy:assets": "cpy 'src/generated/*.json' dist/generated && cpy src/templates/*.hbs dist/templates",

**Comment by sanjaikumarv** [comment]
Just add single quote on src

**Comment by grantwforsythe** [comment]
> Just add single quote on src @adrien-febvay @zhangyahan @yukha-dw , It fixes the issue.
> 
> **Wrong:** npx cpy-cli dist-tmp/** dist/
> 
> **Right:** npx cpy-cli 'dist-tmp/**' dist/

@sanjaikumarv Solution did indeed resolve the issue. I am going to try my hand at resolving this issue given that the project is so small and; hopefully, have a fix up by the end of the week.

**Comment by adrien-febvay** [comment]
> Just add single quote on src @adrien-febvay @zhangyahan @yukha-dw , It fixes the issue.
> 
> **Wrong:** npx cpy-cli dist-tmp/** dist/
> 
> **Right:** npx cpy-cli 'dist-tmp/**' dist/

I used double quotation marks because single ones won't work on PowerShell, but other than that, yes it works. Thank you.

```json
{
  "scripts": {
    "postbuild": "npx cpy-cli \"dist-tmp/**\" dist/"
  }
}
```

**Comment by sindresorhus** [comment]
This is not a bug in cpy-cli. When you run `cpy dist-tmp/**/* dist/` without quotes, your shell (bash, zsh, PowerShell, etc.) expands the glob pattern into individual file paths before cpy-cli ever receives the arguments. This is standard shell behavior that affects all CLI tools.

The solution is to quote the glob pattern:

```sh
cpy 'dist-tmp/**/*' dist/
```

Or with double quotes (works in PowerShell too):

```sh
cpy "dist-tmp/**/*" dist/
```

This is already noted in the help text: `<source> can contain globs if quoted`.
